src/software/thunderscope/robot_diagnostics/robot_info.py

```diff
@@ -186,24 +251,26 @@ def create_vision_pattern_label(self, team_colour, radius):
 
         return label
 
-    def update(self, power_status, error_codes):
+    def update(self, motor_status, power_status, error_codes):
         """
         Receives important sections of RobotStatus proto for this robot and updates widget with alerts
         Checks for
             - Whether breakbeam is tripped
+            - If there are any motor faults
             - If Battery Voltage is too low
             - If this robot has errors
+        :param motor_status: The motor status message for this robot
         :param power_status: The power status message for this robot
         :param error_codes: The error codes of this robot
         :return:
         """
 
-        if power_status.breakbeam_tripped:
-            self.breakbeam_label.setText("In Beam")
-            self.breakbeam_label.setStyleSheet("background-color: red")
-        else:
-            self.breakbeam_label.setText("Not in Beam")
-            self.breakbeam_label.setStyleSheet("background-color: green")
+        self.breakbeam_label.update_breakbeam_status(power_status.breakbeam_tripped)
+
+        self.motor_fault_view.refresh(
+            motor_status,
+            motor_status.front_left.DESCRIPTOR.fields_by_name["motor_faults"],
+        )
```

Comment:
looks like you're only checking front_left

Reply:
I digged around a bit more and it looks like this line is just to get the DESCRIPTOR for the repeated motor fault enum type. It would be a good idea to add a comment clarifying that the line is to get the descriptor rather than the actual field value.

Reply:
yeah this is just to get the descriptor, I'll add a comment explaining that